webserver/vhs-stormforger/README.md

```diff
@@ -0,0 +1,112 @@
+
+1. Record HTTP traffic using VHS.
+2. Configure load test generation container and Optimize experiment.
+3. Run Optimize experiment with load test.
+
+## Capture Data with VHS
+
+The rest of this writeup will assume that you have recorded some network traffic with VHS, and that the recorded network traffic is stored in a Google Cloud Storage object.
+This network traffic should be captured in an environment that is similar or identical to the application that will be optimized.
+The [VHS repository](https://github.com/rename-this/vhs) contains a demonstration that shows how to use VHS to capture http traffic in a kubernetes cluster running the voting web app.
+It can be find in the subdirectory `[hack/demo](https://github.com/rename-this/vhs/tree/main/hack/demo)`.
+This demo deploys the voting web app with a VHS container in each of voting-service pod (4 pods), generates some load with simple script using `curl`, and captures 5 minutes of load to a Google Cloud Storage bucket.
+
+
+### Notes:
+
+- There is a `vhsdemo` service account on GCS that can be used for this demo, or a new one can be created.
+- This demo will produce four output files, one from each voting service pod. We don't currently have a good way to combine these files.
+- The `[loadtest.sh](http://loadtest.sh)` script that's in the repository did not work consistently for me. If you encounter the same issue, try increasing the sleep time on line 9 of the script (The "Wait for the container to come up" sleep line).
+
+## Data Post-processing and Load Test Generation
+
+At this point, there should be four objects in the specified GCS bucket. Each object is a gzipped-json file containing the traffic captured by a VHS instance and will be named according to the session-id of the VHS session that captured it. One of these files must be post-processed to create a Stormforger load test. This process has several steps:
+
+- Generate HAR file using VHS
+- Generate Stormforger load test using `forge`
+- Truncate load test file to manageable size
+
+### Generate HAR file
+
+VHS can read a gzipped JSON file directly from GCS and convert it to a HAR file on your local machine using the following command:
+
+```bash
+vhs --input "gcs|gzip|json --output "har|stdout" --gcs-bucket-name <bucket_name> --gcs-object-name <object_name> --debug > harfile.json
+```
+
+When this command is run, you should see debugging output on the command line indicating that the GCS object is being read and the HAR file is being generated. Once the process is over and the debug output stops, it should be safe to kill this command with ctrl-c. At this point you should have a valid HAR file `harfile.json`.
+
+### Generate Stormforger load test
+
+Given a valid HAR file, it can be easily converted to a Stormforger test in javascript format using the `forge` command line tool.
```

Comment:
```suggestion
Given a valid HAR file, it can be easily converted to a StormForger test in javascript format using the `forge` command line tool.
```